From ecdcaae446fb967155b76cbe771beb5b5a69971d Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 19 Jul 2022 14:49:46 +0800
Subject: [PATCH 20/62] Revert "stm: ltdc: fix two incorrect NULL checks on
 list iterator"

This reverts commit 4e2fbe8cda17d7cfae7429c6d0ac8a11be50cb15.
---
 drivers/gpu/drm/stm/ltdc.c | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/drivers/gpu/drm/stm/ltdc.c b/drivers/gpu/drm/stm/ltdc.c
index 089c00a8e7d4..62488ac14923 100644
--- a/drivers/gpu/drm/stm/ltdc.c
+++ b/drivers/gpu/drm/stm/ltdc.c
@@ -527,8 +527,8 @@ static void ltdc_crtc_mode_set_nofb(struct drm_crtc *crtc)
 	struct drm_device *ddev = crtc->dev;
 	struct drm_connector_list_iter iter;
 	struct drm_connector *connector = NULL;
-	struct drm_encoder *encoder = NULL, *en_iter;
-	struct drm_bridge *bridge = NULL, *br_iter;
+	struct drm_encoder *encoder = NULL;
+	struct drm_bridge *bridge = NULL;
 	struct drm_display_mode *mode = &crtc->state->adjusted_mode;
 	struct videomode vm;
 	u32 hsync, vsync, accum_hbp, accum_vbp, accum_act_w, accum_act_h;
@@ -538,19 +538,15 @@ static void ltdc_crtc_mode_set_nofb(struct drm_crtc *crtc)
 	int ret;
 
 	/* get encoder from crtc */
-	drm_for_each_encoder(en_iter, ddev)
-		if (en_iter->crtc == crtc) {
-			encoder = en_iter;
+	drm_for_each_encoder(encoder, ddev)
+		if (encoder->crtc == crtc)
 			break;
-		}
 
 	if (encoder) {
 		/* get bridge from encoder */
-		list_for_each_entry(br_iter, &encoder->bridge_chain, chain_node)
-			if (br_iter->encoder == encoder) {
-				bridge = br_iter;
+		list_for_each_entry(bridge, &encoder->bridge_chain, chain_node)
+			if (bridge->encoder == encoder)
 				break;
-			}
 
 		/* Get the connector from encoder */
 		drm_connector_list_iter_begin(ddev, &iter);
-- 
2.27.0

